datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -398,7 +434,10 @@ func (f *ChainDataFetcher) handleRequest() {
 				logger.Error("making chain event is failed", "err", err)
 				break
 			}
-			f.handleRequestByType(req.ReqType, req.ShouldUpdateCheckpoint, ev)
+			err = f.handleRequestByType(req.ReqType, req.ShouldUpdateCheckpoint, ev)
+			if err != nil && err == errMaxRetryExceeded {
+				f.forceStop()
```

Comment:
Shouldn't we return here to terminate `handleRequest`?
```suggestion
				f.forceStop()
				return
```

Reply:
Not really. The goroutines are not terminated until the chaindatafetcher is stopped. Here, only fetching/rangeFetching is stopped and it can rerun in the future using API.

Reply:
> Here, only fetching/rangeFetching is stopped and it can rerun in the future using API.

Thanks for your explanation. I think this should be printed as an error log inside `forceStop`.

Reply:
I think the log messages, but outside of `pause`. PTAL @ehnuje.